### ISSUE TITLE
ReferenceError: self is not defined caused by babel-register

### ISSUE THREAD

**Issue by killvung** [jakezatecky/react-checkbox-tree#82]
It seems this react component is now conflicting with babel-register now

```
var crypto = self.crypto || self.msCrypto
             ^
ReferenceError: self is not defined
    at Object.defineProperty.value (/Users/kaldom/p/n/node_modules/react-checkbox-tree/lib/index.js:4695:14)
    at __webpack_require__ (/Users/kaldom/p/n/node_modules/react-checkbox-tree/lib/index.js:31:30)
    at Object.crypto (/Users/kaldom/p/n/node_modules/react-checkbox-tree/lib/index.js:4660:14)
    at __webpack_require__ (/Users/kaldom/p/n/node_modules/react-checkbox-tree/lib/index.js:31:30)
    at Object.<anonymous> (/Users/kaldom/p/n/node_modules/react-checkbox-tree/lib/index.js:1332:15)
    at __webpack_require__ (/Users/kaldom/p/n/node_modules/react-checkbox-tree/lib/index.js:31:30)
    at Object.defineProperty.value (/Users/kaldom/p/n/node_modules/react-checkbox-tree/lib/index.js:1299:18)
    at __webpack_require__ (/Users/kaldom/p/n/node_modules/react-checkbox-tree/lib/index.js:31:30)
    at /Users/kaldom/p/n/node_modules/react-checkbox-tree/lib/index.js:74:18
    at /Users/kaldom/p/n/node_modules/react-checkbox-tree/lib/index.js:77:10
    at webpackUniversalModuleDefinition (/Users/kaldom/p/n/node_modules/react-checkbox-tree/lib/index.js:4:20)
    at Object.<anonymous> (/Users/kaldom/p/n/node_modules/react-checkbox-tree/lib/index.js:11:3)
    at Module._compile (module.js:643:30)
    at Module._extensions..js (module.js:654:10)
    at Object.require.extensions.(anonymous function) [as .js] (/Users/kaldom/p/n/node_modules/babel-register/lib/node.js:152:7)
    at Module.load (module.js:556:32)
```
Also, when I tried to run gulp-mocha 
This shows up 
```
Message:
    self is not defined
Stack:
ReferenceError: self is not defined
    at Object.defineProperty.value (/Users/klsxamm/p/n/node_modules/react-checkbox-tree/lib/index.js:4695:14)
    at __webpack_require__ (/Users/klsxamm/p/n/node_modules/react-checkbox-tree/lib/index.js:31:30)
    at Object.crypto (/Users/klsxamm/p/n/node_modules/react-checkbox-tree/lib/index.js:4660:14)
    at __webpack_require__ (/Users/klsxamm/p/n/node_modules/react-checkbox-tree/lib/index.js:31:30)
    at Object.<anonymous> (/Users/klsxamm/p/n/node_modules/react-checkbox-tree/lib/index.js:1332:15)
    at __webpack_require__ (/Users/klsxamm/p/n/node_modules/react-checkbox-tree/lib/index.js:31:30)
    at Object.defineProperty.value (/Users/klsxamm/p/n/node_modules/react-checkbox-tree/lib/index.js:1299:18)
    at __webpack_require__ (/Users/klsxamm/p/n/node_modules/react-checkbox-tree/lib/index.js:31:30)
    at /Users/klsxamm/p/n/node_modules/react-checkbox-tree/lib/index.js:74:18
    at /Users/klsxamm/p/n/node_modules/react-checkbox-tree/lib/index.js:77:10
```
Has anyone noticed this issue yet?

**Comment by igorroch** [comment]
I've got same result with Server Side Rendering. 
Moving this component to client only resolved issue.

**Comment by killvung** [comment]
This is apparently caused by babel-register which tried compiling the component over the nodejs side

**Comment by jakezatecky** [comment]
Based on some cursory research, the issue with `self` here comes from an assignment of the global level `this`. In the browser environment, the global `this` is `window`. In there, `window.crypto` is defined.

I have no idea what Webpack/Babel is actually using `crypto` for (maybe hashing the imports?), but it appears to not be defined in the Node.js context, likely causing the error we see here. I wonder if changing `libraryTarget` in `webpack.config` to `'commonjs'`, rebuilding the component, and then running your tests would fix it.

**Comment by jonaswindey** [comment]
I've had the same issue. Fixed by downgrading to version 1.02 of react-checkbox-tree
